homeassistant/components/version/__init__.py

```diff
@@ -26,22 +27,33 @@
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up the version integration from a config entry."""
 
-    board = entry.data[CONF_BOARD]
+    source = entry.data[CONF_SOURCE]
 
-    if board not in BOARD_MAP:
+    if (board := entry.data[CONF_BOARD]) not in BOARD_MAP:
         _LOGGER.error(
             'Board "%s" is (no longer) valid. Please remove the integration "%s"',
             board,
             entry.title,
         )
         return False
 
+    if (image := entry.data[CONF_IMAGE]) != DEFAULT_IMAGE and source in (
+        "container",
+        "docker",
+    ):
+        _LOGGER.error(
```

Comment:
As we nowadays have repairs, we should inform the user and help them fix the issue (instead of hiding it in the logs).

../Frenck

Reply:
Thats a completely different PR.